packages/dds/tree/src/feature-libraries/sequence-field/sequenceFieldEditor.ts

```diff
@@ -47,6 +64,80 @@ export const sequenceFieldEditor = {
                   detachedBy: revision,
                   detachIndex,
               }),
+    move(sourceIndex: number, count: number, destIndex: number): Changeset<never> {
+        if (count === 0 || sourceIndex === destIndex) {
+            // TODO: Should we allow creating a move which has no observable effect?
+            return [];
+        }
+
+        const moveOut: Mark<never> = {
+            type: "MoveOut",
+            id: brand(0),
+            count,
+        };
+
+        const moveIn: Mark<never> = {
+            type: "MoveIn",
+            id: brand(0),
+            count,
+        };
+
+        const factory = new MarkListFactory<never>();
+        if (sourceIndex < destIndex) {
+            factory.pushOffset(sourceIndex);
+            factory.pushContent(moveOut);
+            factory.pushOffset(destIndex - sourceIndex);
+            factory.pushContent(moveIn);
+        } else {
+            factory.pushOffset(destIndex);
+            factory.pushContent(moveIn);
+            factory.pushOffset(sourceIndex - destIndex);
+            factory.pushContent(moveOut);
+        }
+        return factory.list;
+    },
+
+    return(
+        sourceIndex: number,
+        count: number,
+        destIndex: number,
+        detachedBy: RevisionTag,
+        detachIndex: number,
+    ): Changeset<never> {
+        if (count === 0) {
+            return [];
+        }
+
+        const id = brand<MoveId>(0);
+        const returnFrom: Mark<never> = {
+            type: "ReturnFrom",
+            id,
+            count,
+            detachedBy,
+        };
+
+        const returnTo: Mark<never> = {
+            type: "ReturnTo",
+            id,
+            count,
+            detachedBy,
+            detachIndex,
+        };
+
+        const factory = new MarkListFactory<never>();
+        if (sourceIndex < destIndex) {
```

Comment:
This doesn't take into account the possibility that `sourceIndex + count > destIndex`. Maybe add a TODO.